packages/std/src/os/vfs.rs

```diff
@@ -55,16 +64,109 @@ pub fn validate_username(username: String) -> Result<bool, ContractError> {
     Ok(true)
 }
 
-pub fn validate_path_name(path: String) -> Result<bool, ContractError> {
-    let re = Regex::new(PATH_REGEX).unwrap();
+pub fn validate_path_name(api: &dyn Api, path: String) -> Result<bool, ContractError> {
+    let andr_addr = AndrAddr::from_string(path.clone());
+    let is_path_reference = path.contains('/');
+    let starts_with_tilde = path.starts_with('~');
+    let includes_protocol = andr_addr.get_protocol().is_some();
+
+    // Path is of the form ~/home/... or ~username/directory
+    if is_path_reference && starts_with_tilde {
+        let re = Regex::new(PATH_REGEX).unwrap();
+        ensure!(
+            re.is_match(&path),
+            ContractError::InvalidPathname {
+                error: Some("Pathname includes an invalid character".to_string())
+            }
+        );
+
+        // let mut components = path.split('/');
+        // let first_component = components.nth(1).unwrap();
+        // ensure!(
+        //    (first_component == "home" || first_component == "lib"),
+        //     ContractError::InvalidPathname {
+        //         error: Some(
+        //             "Paths beginning with ~ must directly reference a username: ~username or root directory: ~/home/username ~/lib/library"
+        //                 .to_string()
+        //         )
+        //     }
+        // );
+
+        return Ok(true);
+    }
 
-    ensure!(
-        re.is_match(&path),
-        ContractError::InvalidPathname {
-            error: Some("Pathname includes an invalid character".to_string())
+    // Path is of the form /home/... or /lib/... or prot://...
+    if is_path_reference && !starts_with_tilde {
```

Comment:
The tests still pass after removing `&& !starts_with_tilde` but maybe it's clearer with it.

Reply:
Yea, was intentionally left in to make it clearer